include/roaring/portability.h

```diff
@@ -437,9 +437,16 @@ static inline int roaring_hamming(uint64_t x) {
 #define croaring_htobe64(x) OSSwapInt64(x)
 
 #elif defined(__has_include) && \
-    __has_include(<byteswap.h>)  // CROARING_IS_BIG_ENDIAN
+    __has_include(              \
+        <byteswap.h>)  && (defined(__linux__) || defined(__FreeBSD__))  // CROARING_IS_BIG_ENDIAN
 #include <byteswap.h>
-#define croaring_htobe64(x) __bswap_64(x)
+#if defined(__linux__)
+#define croaring_htobe64(x) bswap_64(x)
+#elif defined(__FreeBSD__)
+#define croaring_htobe64(x) bswap64(x)
+#else
+#warning "Unknown platform, report as an error"
+#endif
```

Comment:
My report was triggered by reviewing Android NDK r26  builds... whichs use clang and libcxx, and provide `bswap_64`. Based on inspecting some byteswap headers, I implemented this patch:
~~~diff
+#if defined(bswap_64)
+#define croaring_htobe64(x) bswap_64(x)
+#else
 #define croaring_htobe64(x) __bswap_64(x)
+#endif
~~~

Reply:
@dg0yt My expectation is that Android NDK defines __linux__. Is that incorrect?

If so, then the code that I am proposing is correct. Is it not?

Can verify?

Reply:
I can confirm that it builds, so `__linux__` seems to be fine.

Reply:
Great.